HW16/hw16_05_finish.py

```diff
@@ -0,0 +1,139 @@
+class Product:
+
+    def __init__(self, name, price, unit):
+        self.name = name
+        self.price = price
+        self.unit = unit
+
+    def __str__(self):
+        return self.name
+
+    def __float__(self):
+        return self.price / 100
+
+    def __eq__(self, other):
+        return isinstance(other, Product) and self.name == other.name and\
+            self.price == other.price and self.unit == other.unit
+
+    def get_total(self, unit_prod) -> int:
+        total_price: float = self.price * unit_prod / self.unit
+        return int(total_price)
+
+
+class ShoppingCart:
+
+    def __init__(self):
+        self.products = []
+        self.quantities = []
+
+    def __float__(self):
+        return self.get_total() / 100
+
+    def __eq__(self, other):
+        return isinstance(other, ShoppingCart) and self.products == other.products and\
+            self.quantities == other.quantities
+
+    def add_product(self, product, unit_prod) -> None:
+        if product not in self.products:
+            self.products.append(product)
+            self.quantities.append(unit_prod)
+        else:
+            index = self.products.index(product)
+            self.quantities[index] += unit_prod
+
+    def get_total(self) -> int:
+        return int(sum(product.get_total(quantity) for product, quantity in self))
```

Comment:
Метод `Product.get_total` і без того повертає `int`, немає необхідності ще раз робити приведення типів.
```suggestion
        return sum(product.get_total(quantity) for product, quantity in self)
```